request.go

```diff
@@ -125,19 +125,25 @@ func (r *Request) method(method, path string) *Request {
 
 // Body defines the body data to match based on a io.Reader interface.
 func (r *Request) Body(body io.Reader) *Request {
-	r.BodyBuffer, r.Error = ioutil.ReadAll(body)
+	var bodyBuffer []byte
+	bodyBuffer, r.Error = ioutil.ReadAll(body)
+	r.BodyMatchers = append(r.BodyMatchers, bodyBuffer)
 	return r
 }
 
 // BodyString defines the body to match based on a given string.
 func (r *Request) BodyString(body string) *Request {
-	r.BodyBuffer = []byte(body)
+	var bodyBuffer []byte
+	bodyBuffer = []byte(body)
```

Comment:
What about the single-line DRYer version: `bodyBuffer := []byte(body)` ?

Reply:
Or even: `r.BodyMatchers = append(r.BodyMatchers, []byte(body))`

Reply:
Will do. Got a little too zealous with my copy/paste job :)